### PR TITLE
 Fix superadmin access #2307 with user-posted media

### LINKED ISSUE #2307: Superadmin Access Issue in Organizations with User-Posted Media
**Describe the bug**
Superadmin encounters access issues in organizations containing user-posted media, possibly due to a malfunction or disruption caused by the presence of such content.

**To Reproduce**

1. sign in as the user and visit any org 
2. post the content containing media 
3. then sign in as a superAdmin and try to access the same org 
4. you wont be able to do it 

**Expected behavior**
super admin should be able to  open the org and see all the post . 

**Screenshots**

https://github.com/PalisadoesFoundation/talawa-admin/assets/71646466/3169e53b-74e5-4296-9e3e-384b5d5cc560

### PR DESCRIPTION
**Pull request by lokeshsingh3721**
**What kind of change does this PR introduce?**

 bugfix

**Issue Number:**

Fixes #2307 

**Did you add tests for your changes?**

tested locally on the machine

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-api/assets/71646466/53e978cf-0106-49bc-99ad-e69280c2c30b

**If relevant, did you update the documentation?**

No need

**Summary**

There was an issue with organization's post resolver which was not letting the super-admin accessing the organization in which user posted the media.

**Does this PR introduce a breaking change?**

NO

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes 

## Summary by CodeRabbit

- **New Features**
  - Improved post resolver to dynamically update `imageUrl` and `videoUrl` based on the context, ensuring URLs are always up-to-date.

- **Bug Fixes**
  - Enhanced post data formatting and comparison in tests to ensure consistency and accuracy in post-related functionalities.

### CODE REVIEW

**Comment by palisadoes** [comment]
1. Please address the suggestions by Code Rabbit
2. Also increase the test code coverage to an acceptable level

**Comment by lokeshsingh3721** [comment]
> 1. Please address the suggestions by Code Rabbit
> 
>     2. Also increase the test code coverage to an acceptable level

i have made all the changes suggested by Code Rabbit and all the tests are passed .

**Comment by lokeshsingh3721** [comment]
@palisadoes do i need to change anything ?

**Comment by palisadoes** [comment]
> > 1. Please address the suggestions by Code Rabbit
> >    
> >    2. Also increase the test code coverage to an acceptable level
> 
> i have made all the changes suggested by Code Rabbit and all the tests are passed .

There are still 2 code rabbit suggestions that you have not addressed

**Comment by lokeshsingh3721** [comment]
> > > 1. Please address the suggestions by Code Rabbit
> > >    
> > >    2. Also increase the test code coverage to an acceptable level
> > 
> > 
> > i have made all the changes suggested by Code Rabbit and all the tests are passed .
> 
> There are still 2 code rabbit suggestions that you have not addressed

where to see them ?

File: src/resolvers/Organization/posts.ts [hunk 1]
```diff
@@ -75,12 +79,17 @@ export const posts: OrganizationResolvers["posts"] = async (parent, args) => {
       .countDocuments()
       .exec(),
   ]);
+  const posts = objectList.map((post: InterfacePost) => ({
+    ...post,
+    imageUrl: post.imageUrl ? `${context.apiRootUrl}${post.imageUrl}` : null,
+    videoUrl: post.videoUrl ? `${context.apiRootUrl}${post.videoUrl}` : null,
```
[the remark this answers is not in the document]
Comment (lokeshsingh3721) [reply src/resolvers/Organization/posts.ts]: @palisadoes  are you talking about these two ?

[the remark this answers is not in the document]
Comment (palisadoes) [reply src/resolvers/Organization/posts.ts]: Please address thi

[the remark this answers is not in the document]
Comment (palisadoes) [reply src/resolvers/Organization/posts.ts]: Please address this

File: tests/resolvers/Organization/posts.spec.ts [hunk 2]
```diff
@@ -71,12 +70,66 @@ describe("resolvers -> Organization -> post", () => {
...
+        },
+      ],
+      pageInfo: {
+        endCursor: testPost?._id.toString(),
+        hasNextPage: false,
+        hasPreviousPage: false,
+        startCursor: testPost2?._id.toString(),
+      },
+      totalCount,
+    });
```
[the remark this answers is not in the document]
Comment (palisadoes) [reply tests/resolvers/Organization/posts.spec.ts]: Please address this

File: src/resolvers/Organization/posts.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (palisadoes) [reply src/resolvers/Organization/posts.ts]: @lokeshsingh3721 Please address this

[the remark this answers is not in the document]
Comment (lokeshsingh3721) [reply src/resolvers/Organization/posts.ts]: > @lokeshsingh3721 Please address this

image
I am sorry for stretching  this out but i think  I have already made these changes and i am not seeing any other suggestions . Can you please point out or share screenshots of another changes ?

**Review by palisadoes — APPROVED** [review]

**Pull request merged**